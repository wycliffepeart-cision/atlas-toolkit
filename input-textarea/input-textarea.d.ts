export declare class InputTextarea extends HTMLElement {
    static get observedAttributes(): string[];
    attributeChangedCallback(name: any, _oldValue: any, newValue: any): void;
    get value(): string;
    set value(val: string);
    get disabled(): boolean;
    set disabled(value: boolean);
}
//# sourceMappingURL=input-textarea.d.ts.map