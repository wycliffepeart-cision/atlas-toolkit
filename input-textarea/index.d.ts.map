{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../src/input-textarea/index.ts"],"names":[],"mappings":""}