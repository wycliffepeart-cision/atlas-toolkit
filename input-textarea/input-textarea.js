import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var InputTextareaTemplate = "<template>\n  <textarea id=\"Input\" part=\"input\"></textarea>\n\n  <label for=\"Input\" part=\"label-wrapper\">\n    <slot name=\"label\" part=\"label\"></slot>\n    <slot name=\"required\" part=\"required\">*</slot>\n  </label>\n\n  <slot name=\"helperText\" part=\"helper-text\"></slot>\n  <slot name=\"validationText\" part=\"validation-text\"></slot>\n\n  <slot part=\"content\"></slot>\n</template>\n";

const TEXTAREA_ATTRIBUTES = ['disabled', 'maxlength', 'placeholder', 'required', 'readonly', 'name', 'value'];
let InputTextarea = class InputTextarea extends HTMLElement {
    static get observedAttributes() {
        return TEXTAREA_ATTRIBUTES;
    }
    attributeChangedCallback(name, _oldValue, newValue) {
        if (TEXTAREA_ATTRIBUTES.includes(name.trim().toLowerCase())) {
            const input = this.shadowRoot.getElementById('Input');
            if (newValue === null) {
                input.removeAttribute(name);
            }
            else {
                input.setAttribute(name, newValue);
            }
        }
        // unknown attribute?
    }
    get value() {
        const inputElem = this.shadowRoot.getElementById('Input');
        return inputElem.value;
    }
    set value(val) {
        const inputElem = this.shadowRoot.getElementById('Input');
        inputElem.value = val;
    }
    get disabled() {
        const dis = this.getAttribute('disabled');
        return dis !== null && dis !== 'false';
    }
    set disabled(value) {
        this.toggleAttribute('disabled', value);
    }
};
InputTextarea = __decorate([
    Template(InputTextareaTemplate)
], InputTextarea);

export { InputTextarea };
