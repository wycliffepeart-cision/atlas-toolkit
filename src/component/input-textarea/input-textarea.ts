import { Template } from '../../decorators/template';
import InputTextareaTemplate from './input-textarea.html';

const TEXTAREA_ATTRIBUTES = ['disabled', 'maxlength', 'placeholder', 'required', 'readonly', 'name', 'value'];

@Template(InputTextareaTemplate)
export class InputTextarea extends HTMLElement {
  static get observedAttributes() {
    return TEXTAREA_ATTRIBUTES;
  }

  attributeChangedCallback(name, _oldValue, newValue) {
    if (TEXTAREA_ATTRIBUTES.includes(name.trim().toLowerCase())) {
      const input = this.shadowRoot.getElementById('Input');
      if (newValue === null) {
        input.removeAttribute(name);
      } else {
        input.setAttribute(name, newValue);
      }
    }

    // unknown attribute?
  }

  get value() {
    const inputElem = this.shadowRoot.getElementById('Input') as HTMLInputElement;
    return inputElem.value;
  }

  set value(val) {
    const inputElem = this.shadowRoot.getElementById('Input') as HTMLInputElement;
    inputElem.value = val;
  }

  get disabled() {
    const dis = this.getAttribute('disabled');
    return dis !== null && dis !== 'false';
  }

  set disabled(value) {
    this.toggleAttribute('disabled', value);
  }
}
