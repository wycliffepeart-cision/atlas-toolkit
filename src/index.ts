export * from './component/accordion/accordion';
export * from './component/alert/alert';
export * from './component/application-frame/application-frame';
export * from './component/button/button';
export * from './component/icon/icon';
export * from './component/button-set/button-set';
export * from './component/chip/chip';
export * from './component/chip-group/chip-group';
export * from './component/dialog/dialog';
export * from './component/feed/feed';
export * from './component/input-checkbox/input-checkbox';
export * from './component/input-chip-group/input-chip-group';
export * from './component/input-dropdown/input-dropdown';
export * from './component/input-search/input-search';
export * from './component/input-text/input-text';
export * from './component/input-textarea/input-textarea';
export * from './component/logo/logo';
export * from './component/main-nav/main-nav';
export * from './component/main-nav-item/main-nav-item';
export * from './component/number-display/number-display';
export * from './component/profile-control/profile-control';
export * from './component/snackbar/snackbar';
export * from './component/stepper/stepper';
export * from './component/stepper-step/stepper-step';
export * from './component/tile-board/tile-board';
export * from './component/tile-board-tile/tile-board-tile';
export * from './component/workflow/workflow';
export * from './component/workflow-step/workflow-step';
