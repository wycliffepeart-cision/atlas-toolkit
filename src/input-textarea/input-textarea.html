<template>
  <textarea id="Input" part="input"></textarea>

  <label for="Input" part="label-wrapper">
    <slot name="label" part="label"></slot>
    <slot name="required" part="required">*</slot>
  </label>

  <slot name="helperText" part="helper-text"></slot>
  <slot name="validationText" part="validation-text"></slot>

  <slot part="content"></slot>
</template>
