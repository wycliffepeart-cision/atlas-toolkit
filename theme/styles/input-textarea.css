@layer atlas {
  atlas-input-textarea {
    --minimum-height: 7.25em;

    --disabled-background-color: var(--atlas-color-primary-n50);

    --padding-block: 0.5em;
    --padding-inline: 1em;

    --padding-block-start-label-spacer: calc(1em + 0.125em);
    --padding-block-start-calced: calc(var(--padding-block) + var(--padding-block-start-label-spacer));
    --total-block-padding: calc(var(--padding-block) + var(--padding-block-start-calced));
    --minimum-height-calced: calc(var(--minimum-height) - var(--total-block-padding));
  }

  atlas-input-textarea {
    position: relative;

    accent-color: var(--atlas-input-common-accent-color);

    display: grid;
    margin-block: var(--atlas-input-common-block-margin);
  }

  atlas-input-textarea::part(input) {
    display: block;
    min-height: var(--minimum-height-calced);

    font-size: inherit;
    font-family: inherit;

    padding-block-start: var(--padding-block-start-calced);
    padding-block-end: var(--padding-block);
    padding-inline: var(--padding-inline);

    color: var(--atlas-color-primary-n400);

    background-color: var(--atlas-input-common-background-color);
    border-radius: var(--atlas-input-common-border-radius);
    border: var(--atlas-input-common-border);
  }

  atlas-input-textarea[disabled]:not([disabled='false'])::part(input) {
    background-color: var(--disabled-background-color);
    border-color: var(--disabled-background-color);
  }

  atlas-input-textarea::part(input):focus {
    outline: none;
    box-shadow: var(--atlas-input-common-box-shadow);
  }

  atlas-input-textarea::part(label-wrapper) {
    position: absolute;
    top: calc(var(--padding-block) + 0.125em);
    left: var(--padding-inline);

    color: var(--atlas-color-primary-n300);
  }

  atlas-input-textarea::part(label) {
    font-size: var(--atlas-input-common-label-font-size);
  }

  atlas-input-textarea::part(required) {
    font-size: var(--atlas-input-common-label-font-size);
    display: none;
  }

  atlas-input-textarea[required]:not([required='false'])::part(required) {
    display: inline-block;
  }

  atlas-input-textarea::part(helper-text) {
    display: none;
  }

  atlas-input-textarea::part(content) {
    display: none;
  }
}
