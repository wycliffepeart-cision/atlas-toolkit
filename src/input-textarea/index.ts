import { InputTextarea } from './input-textarea';

if (!customElements.get('atlas-input-textarea')) customElements.define('atlas-input-textarea', InputTextarea);
