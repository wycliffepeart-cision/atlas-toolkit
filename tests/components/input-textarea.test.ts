import '../../src/input-textarea';

const INPUT_TEXTAREA_TAG = 'atlas-input-textarea';

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();
});

describe('Input', () => {
  describe('Textarea JS API', () => {
    describe('.value', () => {
      it('getter', async () => {
        await customElements.whenDefined(INPUT_TEXTAREA_TAG);
        const search: any = document.createElement(INPUT_TEXTAREA_TAG);
        (expect(search.value) as any).toEqual('');
      });

      it('setter', async () => {
        await customElements.whenDefined(INPUT_TEXTAREA_TAG);
        const search: any = document.createElement(INPUT_TEXTAREA_TAG);
        search.value = 'SOME FAKE STRING';

        (expect(search.value) as any).toEqual('SOME FAKE STRING');
      });

      it('from value attribute', async () => {
				await customElements.whenDefined(INPUT_TEXTAREA_TAG);
				const search: any = document.createElement(INPUT_TEXTAREA_TAG);
        search.setAttribute('value', 'SOME FAKE STRING');

        (expect(search.value) as any).toEqual('');
      });
    });

    describe('.disabled', () => {
      it('getter (default)', async () => {
				await customElements.whenDefined(INPUT_TEXTAREA_TAG);
				const search: any = document.createElement(INPUT_TEXTAREA_TAG);

        (expect(search.disabled) as any).toEqual(false);
      });

      it('setter (identity)', async () => {
				await customElements.whenDefined(INPUT_TEXTAREA_TAG);
				const search: any = document.createElement(INPUT_TEXTAREA_TAG);

        search.disabled = true;
        (expect(search.disabled) as any).toEqual(true);
      });

      it('from toggle attribute - true', async () => {
				await customElements.whenDefined(INPUT_TEXTAREA_TAG);
				const search: any = document.createElement(INPUT_TEXTAREA_TAG);

        search.toggleAttribute('disabled', true);
        (expect(search.disabled) as any).toEqual(true);
      });

      it('from non-toggle attribute - false', async () => {
				await customElements.whenDefined(INPUT_TEXTAREA_TAG);
				const search: any = document.createElement(INPUT_TEXTAREA_TAG);

        search.setAttribute('disabled', 'false');
        (expect(search.disabled) as any).toEqual(false);
      });

      it('from non-toggle attribute - string', async () => {
				await customElements.whenDefined(INPUT_TEXTAREA_TAG);
				const search: any = document.createElement(INPUT_TEXTAREA_TAG);

        search.setAttribute('disabled', 'SOME_FAKE_VALUE');
        (expect(search.disabled) as any).toEqual(true);
      });
    });
  });

  describe('Textarea', () => {
    it('exists', async () => {
      await customElements.whenDefined(INPUT_TEXTAREA_TAG);
      const dashboardElement = document.createElement(INPUT_TEXTAREA_TAG);
      document.body.appendChild(dashboardElement);

      (expect(dashboardElement) as any).toBeInTheDocument();
    });

    it('is matches snapshot', () => {
      const elem = document.createElement(INPUT_TEXTAREA_TAG);
      document.body.appendChild(elem);

      (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
    });

    it('is matches snapshot when disabled', () => {
      const elem = document.createElement(INPUT_TEXTAREA_TAG);
      document.body.appendChild(elem);

      elem.toggleAttribute('disabled', true);

      (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
    });

    it('is matches snapshot when disabled toggled on/off', () => {
      const elem = document.createElement(INPUT_TEXTAREA_TAG);
      document.body.appendChild(elem);

      elem.toggleAttribute('disabled', true);
      elem.toggleAttribute('disabled', false);

      (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
    });
  });
});
