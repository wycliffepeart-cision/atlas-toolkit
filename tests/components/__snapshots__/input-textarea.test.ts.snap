// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Textarea is matches snapshot 1`] = `
<textarea id="Input"
          part="input"
>
</textarea>
<label for="Input"
       part="label-wrapper"
>
  <slot name="label"
        part="label"
  >
  </slot>
  <slot name="required"
        part="required"
  >
    *
  </slot>
</label>
<slot name="helperText"
      part="helper-text"
>
</slot>
<slot name="validationText"
      part="validation-text"
>
</slot>
<slot part="content">
</slot>
`;

exports[`Input Textarea is matches snapshot when disabled 1`] = `
<textarea id="Input"
          part="input"
          disabled
>
</textarea>
<label for="Input"
       part="label-wrapper"
>
  <slot name="label"
        part="label"
  >
  </slot>
  <slot name="required"
        part="required"
  >
    *
  </slot>
</label>
<slot name="helperText"
      part="helper-text"
>
</slot>
<slot name="validationText"
      part="validation-text"
>
</slot>
<slot part="content">
</slot>
`;

exports[`Input Textarea is matches snapshot when disabled toggled on/off 1`] = `
<textarea id="Input"
          part="input"
>
</textarea>
<label for="Input"
       part="label-wrapper"
>
  <slot name="label"
        part="label"
  >
  </slot>
  <slot name="required"
        part="required"
  >
    *
  </slot>
</label>
<slot name="helperText"
      part="helper-text"
>
</slot>
<slot name="validationText"
      part="validation-text"
>
</slot>
<slot part="content">
</slot>
`;
